indigo/docs/guides/howto-hello-indigo-tyrian.md

```diff
@@ -0,0 +1,418 @@
+  "staticOutPath": "assets",
+  "watcherGlob": "**"
+}
+```
+
+We need to add a couple of dependencies. Firstly we'll add the
+dependency for static files by running
+`yarn add parcel-reporter-static-files-copy --dev` from the command line; and
+secondly we'll add `snabbdom` for Tyrian by running `yarn add snabbdom`.
+
+Finally, we need to configure ParcelJS so that it knows how to copy our static
+files. Add a `.parcelrc` file and add the following:
+
+```json
+{
+  "extends": ["@parcel/config-default"],
+  "reporters":  ["...", "parcel-reporter-static-files-copy"]
+}
+```
+
+### Update HelloIndigo.scala
+
+We'll be using the `TyrianSubSystem` as a way of communicating between Indigo
+and Tyrian. Add `import tyrian.TyrianSubSystem` to the imports in
+`HelloIndigo.scala`, and then update the `object` to be a `case class` so that
+it it contains the subsystem as an argument. We can also remove the top level
+export, as we'll no longer need it:
+
+```diff
+- import scala.scalajs.js.annotation.JSExportTopLevel
++ import tyrian.TyrianSubSystem
+
+- @JSExportTopLevel("IndigoGame")
+- object HelloIndigo extends IndigoGame[Unit, Unit, Model, Unit] {
++ final case class HelloIndigo(tyrianSubSystem: TyrianSubSystem[Int]) extends IndigoGame[Unit, Unit, Model, Unit] {
+```
+
+We'll also need to tell Indigo to use the new sub-system, which we can do by
+adding an entry to the `boot` method like so:
+
+```diff
+   def boot(flags: Map[String, String]): Outcome[BootResult[Unit]] =
+     Outcome(
+       BootResult
+         .noData(config)
+         .withAssets(assets)
++        .withSubSystems(tyrianSubSystem)
+     )
+```
+
+### Create a Tyrian App
+
+We'll now develop our initial Tyrian app, which will consist simply of
+a canvas (for Indigo), a counter, and a button, wrapped in a few `div` elements.
+Create a new file called `HelloTyrian.scala` inside the `helloindigo/src` folder
+and add the following contents found
+[here](https://gist.github.com/hobnob/436318b3ae5eed5891ba2b18bb8c264b).
+
+You can now build the project in mill or sbt and then run `yarn start` to see the HelloIndigo demo running inside a Tyrian website. This is great if you want a
+website surrounding your game with no interaction, but it would be much more
+useful to get them both talking to each other... this is what we're doing next!
+
+## Tyrian to Indigo Communication
+
+The first thing we'll do is get Tyrian to request that Indigo add a new dot. To
+do this we'll need to add an `OnClick` event to our button, which will tell
+Indigo what to do.
+
+In `HelloTyrian.scala`, we'll need a new message type. Add a new message type called AddDot like so:
+
+```diff
+   enum Msg:
+     case StartIndigo extends Msg
++    case AddDot      extends Msg
+```.
+
+Next we'll need to deal with that message in Tyrian, by adding a case to `update`
+like this:
+
+```diff
+     msg match
++      case Msg.AddDot =>
++        (model, model.bridge.publish(IndigoGameId(gameDivId), 0))
```

Comment:
Technically, you don't need the `IndigoGameId(gameDivId)` stuff since you only have one game - omitting and ID means "publish to all". Might be cleaner without it?

Reply:
So just `model.bridge.publish(0)`?